File: CodeChef/Beginner/08SmallFactorials/smallfactorials.py
class smallfactorials:
    def fact(self, x):
        for i in range(2, x):
            x *= i
        return x

    def main(self):
        n = int(input())
        for i in range(n):
            x = int(input())
            print(self.fact(x))


if __name__ == "__main__":
    smallfactorials().main()

File: CodeChef/Beginner/08SmallFactorials/smallfactorials.java
import java.util.Scanner;
import java.math.BigInteger;

class smallfactorials {

    public static BigInteger fact(int x) {

        // BigInteger bint = new BigInteger("1");
        // BigInteger bint =BigInteger.ONE

        BigInteger bint = BigInteger.valueOf(1);
        for (int i = 2; i <= x; i++) {
            bint = bint.multiply(BigInteger.valueOf(i));
        }
        return bint;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n, x;
        n = sc.nextInt();
        for (int i = 0; i < n; i++) {
            x = sc.nextInt();
            System.out.println(fact(x));

        }
    }
}
